src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -223,6 +224,22 @@ private void CheckFileSystem()
             }
         }
 
+        private void ValidateLinuxSKUConfiguration()
+        {
```

Comment:
The logs from repro app shows that the indicator of error is below line - https://github.com/Azure/azure-functions-host/blob/88b610c61a6dc39e053f5fe7f902204e06aa1b41/src/WebJobs.Script.WebHost/Configuration/ScriptApplicationHostOptionsSetup.cs#L97

Added similar check here to identify if the app has - zip deploy + MSI + Linux Consumption to log the error and raise the diagnostic event.